src/main/java/com/cronutils/model/time/ExecutionTime.java

```diff
@@ -373,7 +394,25 @@ public Duration timeFromLastExecution(ZonedDateTime date){
      * @return true if date matches cron expression requirements, false otherwise.
      */
     public boolean isMatch(ZonedDateTime date){
-        return nextExecution(lastExecution(date)).equals(date);
+        ZonedDateTime lastExecTime = lastExecution( date ) ;
+        if( lastExecTime != null ) {
+            ZonedDateTime nextExecTime = nextExecution( lastExecTime ) ;
+            if( nextExecTime != null ) {
+               return nextExecTime.equals( date ) ; 
+            }
+        }
+        
+        int year = date.getYear() ;
```

Comment:
All the if conditions could be refactored to a single return statement.
However, since this statement will nevertheless not be trivial, it should be in a separate method and this method should be tested.